src/main/java/com/exasol/spark/s3/ExasolS3Table.java

```diff
@@ -58,12 +64,57 @@ public Set<TableCapability> capabilities() {
 
     @Override
     public ScanBuilder newScanBuilder(final CaseInsensitiveStringMap map) {
-        return null; // this will be implemented in #149
+        final ExasolOptions options = getExasolOptions(map);
+        validateNumberOfPartitions(options);
+        updateSparkConfigurationForS3(options);
+        return new ExasolS3ScanBuilder(options, this.schema, map);
     }
 
     @Override
     public WriteBuilder newWriteBuilder(final LogicalWriteInfo defaultInfo) {
         return null; // this will be implemented in #149
     }
 
+    private ExasolOptions getExasolOptions(final CaseInsensitiveStringMap options) {
+        final ExasolOptions.Builder builder = ExasolOptions.builder() //
+                .jdbcUrl(options.get(JDBC_URL)) //
+                .username(options.get(USERNAME)) //
+                .password(options.get(PASSWORD)) //
+                .s3Bucket(options.get(S3_BUCKET));
+        if (options.containsKey(TABLE)) {
+            builder.table(options.get(TABLE));
+        } else if (options.containsKey(QUERY)) {
+            builder.query(options.get(QUERY));
+        }
+        return builder.withOptionsMap(options.asCaseSensitiveMap()).build();
+    }
+
+    private void validateNumberOfPartitions(final ExasolOptions options) {
+        final int numberOfPartitions = options.getNumberOfPartitions();
+        if (numberOfPartitions > MAX_ALLOWED_NUMBER_OF_PARTITIONS) {
+            throw new ExasolValidationException(ExaError.messageBuilder("E-SEC-23")
+                    .message("The number of partitions is larger than maximum allowed {{MAXPARTITIONS}} value.",
+                            String.valueOf(MAX_ALLOWED_NUMBER_OF_PARTITIONS))
+                    .mitigation("Please set the number of partitions parameter to a lower value.").toString());
+        }
+    }
+
+    private void updateSparkConfigurationForS3(final ExasolOptions options) {
+        final SparkSession sparkSession = SparkSession.active();
+        synchronized (sparkSession.sparkContext().hadoopConfiguration()) {
+            final Configuration conf = sparkSession.sparkContext().hadoopConfiguration();
+            if (options.hasEnabled(CI_ENABLED)) {
+                conf.set("fs.s3a.aws.credentials.provider", "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider");
+                conf.set("fs.s3a.access.key", options.get(AWS_ACCESS_KEY_ID));
+                conf.set("fs.s3a.secret.key", options.get(AWS_SECRET_ACCESS_KEY));
+            }
+            if (options.containsKey(S3_ENDPOINT_OVERRIDE)) {
+                conf.set("fs.s3a.endpoint", "http://" + options.get(S3_ENDPOINT_OVERRIDE));
+            }
+            if (options.hasEnabled(S3_PATH_STYLE_ACCESS)) {
+                conf.set("fs.s3a.path.style.access", "true");
+            }
+        }
+    }
```

Comment:
I propose to try to keep the name "CI_ENABLED" out of the production code and rather move the logic to the setup of the CI test.
In the current place I propose to define 3 additional options and generous documentation, e.g.
- `FS_S3A_AWS_CREDENTIALS_PROVIDER`
- `FS_S3A_ACCESS_KEY`
- `FS_S3A_ACCESS_SECRET`

Another option could be to support an option like `AWS_CREDENTIALS_PROVIDER` and for CI tests set its value to `"org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider"`.
If  class `ExasolS3Table` then finds this option set, then class `ExasolS3Table` could
```
  conf.set("fs.s3a.aws.credentials.provider", options.get(AWS_CREDENTIALS_PROVIDER));
  conf.set("fs.s3a.access.key", options.get(AWS_ACCESS_KEY_ID));
  conf.set("fs.s3a.secret.key", options.get(AWS_SECRET_ACCESS_KEY));
```

And we could move the constants out of class `Constants` and convert them into enums, see branch [feature/#149-proposals_ck](https://github.com/exasol/spark-connector/tree/feature/%23149-proposals_ck).

Reply:
Please feel free to discuss. 🙂

Reply:
These great proposals, I am going to address them

Reply:
Added